src/core/edgeNetwork/mergeLifecycleResponses.js

```diff
@@ -18,10 +19,25 @@ export default returnValues => {
   const consumerOnResponseReturnValues = returnValues.shift() || [];
   const lifecycleOnBeforeRequestReturnValues = returnValues;
 
-  return assign(
-    {},
+  return [
     ...lifecycleOnResponseReturnValues,
     ...consumerOnResponseReturnValues,
     ...lifecycleOnBeforeRequestReturnValues
-  );
+  ].reduce((accumulator, currentValue) => {
```

Comment:
Would it help to extract this code into a separate util and unit test that alone?